gping: init at 1.1

andrew-d opened the pull request:
###### Motivation for this change
[gping](https://github.com/orf/gping) is a utility to display a graph of ping timing in a terminal.  It looks like this:
![screencast](https://github.com/orf/gping/raw/master/doc/readme_screencast.gif)

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

andrew-d committed:
gping: init at 1.1

gping is a utility to display a graph of ping timing in a terminal

andrew-d commented on pkgs/tools/networking/gping/default.nix:
```diff
@@ -0,0 +1,24 @@
+{ pkgs
```
Bleh, sorry - bad commit; forced-pushed an fix.

andrew-d commented on pkgs/tools/networking/gping/default.nix:
```diff
@@ -0,0 +1,24 @@
+
+  propagatedBuildInputs = with pythonPackages; [ colorama ];
+
+  src = pkgs.fetchurl {
```
Thanks! Didn't know about that function - switched to using that.

andrew-d commented on pkgs/tools/networking/gping/default.nix:
```diff
@@ -0,0 +1,24 @@
+}:
+
+pythonPackages.buildPythonApplication rec {
+  name = "gping-${version}";
```
Done!

nlewo commented on pkgs/tools/networking/gping/default.nix:
```diff
@@ -0,0 +1,25 @@
+{ stdenv
+, lib
+
```
I don't think this blankline is usefull

andrew-d replied:
Fixed!

andrew-d commented:
@nlewo - Okay, fixed the blank line. I also updated to use Python 3, since it uses `subprocess.getoutput` in one place, which is only available in Python 3. And finally, I patched the path to the `ping` binary. This means it won't pick up the `setuid` wrapper under `/run/wrappers/bin`, but I like this approach better (and one can always add a wrapper for the `gping` binary).

nlewo commented:
Thx!
